bundles/org.openhab.binding.astro/src/main/java/org/openhab/binding/astro/internal/model/Eclipse.java

```diff
@@ -12,52 +12,73 @@
  */
 package org.openhab.binding.astro.internal.model;
 
+import java.util.AbstractMap.SimpleEntry;
 import java.util.Calendar;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
 
-import org.apache.commons.lang.builder.ToStringBuilder;
-import org.apache.commons.lang.builder.ToStringStyle;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
 
 /**
  * Holds eclipse informations.
  *
  * @author Gerhard Riegler - Initial contribution
  */
+@NonNullByDefault
 public class Eclipse {
-    private Calendar total;
-    private Calendar partial;
+    private final Map<EclipseKind, Entry<Calendar, @Nullable Double>> entries = new HashMap<>();
```

Comment:
To make the null checks more consistent, you should just make the map value nullable here.
```suggestion
    private final Map<EclipseKind, @Nullable Entry<Calendar, @Nullable Double>> entries = new HashMap<>();
```